Various/BuyOne_Convert custom action to LUA ReaScript.lua

```diff
@@ -0,0 +1,313 @@
+			elseif v == "2002" then str = go_loop_start;
+				-- Insert repeat at the very start of action sequence if loop start wasn't set explicitly by 2001
+				if not repeat_loop then table.insert(code_t, 1, set_loop_start) end
+			elseif tonumber(v) and (tonumber(v) > 2007 and tonumber(v) < 2013) and sect_media_main == '0' then str = wait(v)
+			else str = 'r.Main_OnCommand('..func..',0) -- '..r.CF_GetCommandText(tonumber(sect_media_main), id) end
+		end
+	code_t[#code_t+1] = str
+	end
+
+
+-- Concatenate prefix, comments and addiditional code
+local pref, inset1, inset2, inset3 = '','','',''
+local sect_code = code:match('%s(3206%d)%s\"'..comm_id..'')
+	if sect_code == '32060' then pref = 'MIDI Ed_'
+	elseif sect_code == '32061' then pref = 'MIDI EvL_'
+	elseif sect_code == '32063' then pref = 'Media Ex_'
+	else pref = 'Main_' end
+
+
+	if midi_from_arrange == 'a' and (sect_code == '32060' or sect_code == '32061') then inset1, inset2, inset3, pref = '\n-- Import into the Main section of the Action list and run from Arrange.', 'r.Main_OnCommand(40153, 0) -- Item: Open in built-in MIDI editor\n\nlocal hwnd = r.MIDIEditor_GetActive()\n\n','\n\nr.MIDIEditor_OnCommand(hwnd,2) -- File: Close window', pref..'from Arrange_'
+	else
+		if sect_code == '32060' then inset2, sect_name = 'local hwnd = r.MIDIEditor_GetActive()\n\n', 'MIDI Editor and/or MIDI Event list editor'; loc = 'inside '..sect_name -- loc is separate because it uses previously initialized variable
+		elseif sect_code == '32061' then inset2, sect_name = 'local hwnd = r.MIDIEditor_GetActive()\n\n', 'MIDI Event list editor and/or MIDI Editor'; loc = 'inside '..sect_name
+		elseif sect_code == '32063' then sect_name = 'Media Explorer'; loc = 'inside '..sect_name
+		else sect_name = 'Main / Main (alt recording)'; loc = 'Arrange'
+		end
+	inset1 = '\n-- Import into the '..sect_name..' section of the Action list and run from '..loc..'.'
+	end
+
+
+-- Concatenate LUA file code
+local output = '-- Converted from a custom action \"Custom: '..cust_act_name..'\"'..inset1..'\n\n\nlocal r = reaper\n\nr.PreventUIRefresh(1)\nr.Undo_BeginBlock();\n\n'..inset2..table.concat(code_t,'\n')..inset3..'\n\nr.Undo_EndBlock(\"'..cust_act_name..'\",-1)\nr.PreventUIRefresh(-1)\n\n\n'
+
+local f = io.open(f_path..sep..pref..'CA_'..f_name..'.lua','w')
+local result,_,_ = f:write(output)
+f:close()
+
+local mess, head, mode = table.unpack(result and {'       File has been created successfully\n\n      and placed in the \\Scripts'..subdir_txt..'  folder\n\n       of the REAPER program directory.\n\nWould you like to open the subfolder now?', 'SUCCESS', 4} or {'File creation has failed.', 'FAILURE', 0}) -- idiom source https://stackoverflow.com/questions/25362528/lua-ternary-operator-multiple-variables
+
+local resp = r.MB(mess,head,mode)
+
+local command = sep == '\\' and 'explorer ' or 'open ' -- Win or Mac
+
+	if resp == 6 then os.execute(command..f_path) end
```

Comment:
This is vulnerable to command injection. Either the `f_path` variable should be sanitized or `CF_LocateInExplorer`/`CF_ShellExecute` from SWS should be used instead.

Reply:
I'm not completely sure what sanitization means and its purpose. On line 186 i tried to make sure that no illegal characters feature in the file name `local f_name = cust_act_name:gsub('[\\/:*?\"<>|]', '')`. Is this it?

If it poses security risk should it also be sanitized immediately before making a call via shell or your function? And if so what goes into such sanitization, any other characters?

Reply:
I'm not completely sure what sanitization means and its purpose. Is this removal of illegal characters similar to what's done on line 186 i tried to make sure that no illegal characters feature in the file name `local f_name = cust_act_name:gsub('[\\/:*?\"<>|]', '')`  or is it something else?

Reply:
I'm not completely sure what sanitization means and its purpose. Is this removal of illegal characters similar to what's done on line 186 `local f_name = cust_act_name:gsub('[\\/:*?\"<>|]', '')`  or is it something else?

Reply:
I'm not completely sure what sanitization means and its purpose. Is this removal of illegal characters similar to what's done on line 186 `local f_name = cust_act_name:gsub('[\\/:*?\"<>|]', '')`  or is it something else? I don't mind using your functions, just want to understand what goes into sanitization.

Reply:
I'm not completely sure what sanitization means and its purpose. Is this removal of illegal characters similar to what's done on line 186 `local f_name = cust_act_name:gsub('[\\/:*?\"<>|]', '')`  or is it making it innocuous? I don't mind using your functions, just want to understand what goes into sanitization.
Found Lokasenna's function:

`local function sanitize_filename(name)
    return string.gsub(name, "[^%w%s]", "-")
end`

Will this do?

Reply:
I'm not completely sure what sanitization means and its purpose. Is this removal of illegal characters similar to what's done on line 186 `local f_name = cust_act_name:gsub('[\\/:*?\"<>|]', '')`  or is it making it innocuous? I don't mind using your functions, just want to understand what goes into sanitization.